content/post/debrick-core9g25-using-sam-ba-on-linux.md

```diff
@@ -0,0 +1,286 @@
+Syntax:
+    write:<filename>:[<addr>]
+Examples:
+    write:firmware.bin         write firmware.bin to start of memory
+    write:firmware.bin:0x1000  write firmware.bin at offset 0x1000
+
+* writeboot - write boot program to memory from a file
+Syntax:
+    writeboot:<filename>
+Example:
+    writeboot:firmware.bin  write firmware.bin as boot program at start of memory
+```
+
+## Serial connection
+Connect your Linux machine to the serial debug port of the device. I've used
+[this USB-to-serial cable][usb-to-serial].
+
+Before I continue I often use Minicom to validate if the serial connection is
+working. Start Minicom and configure the serial port with the settings below.
+You can open the settings menu by pressing `Ctrl-a` and then `o` (the letter,
+not the number).
+
+* baudrate: 115200
+* parity: none
+* bytesize: 8
+* stopbits: 1
+
+{{<figure src="/img/configure_minicom.png">}}
+
+Reboot the CPU. When you see human readable data in Minicom the serial port
+has been configured correctly. You probably should at least see the string
+"romboot".
+
+If you don't see anything, you probably need to switch your Rx and Tx lines.
+
+When Minicom prints garbage it's likely that the settings on both ends of the
+serial line doesn't match.
+
+## Boot from ROM
+If your serial connection has been set up it's time to fix debrick the NAND.
+The ROM of the AT91SAM9G25 contains a program which is also called SAM-BA,
+SAM-BA on your Linux machine interacts with SAM-BA on the ROM via the serial
+connection.
+
+In order to boot from ROM the BMS pin must be `1`. The BMS pin can be brought
+`1` by shortening the marked 2 island on the picture below.
+
+{{<figure src="/img/core9g25_with_marked_bms.jpg">}}
+
+Before you continue make sure that you've closed Minicom! Only 1 program at the
+time can use serial line.
+
+## Initialize RAM
+Enable the BMS pin while rebooting the device; the device will boot SAM-BA program
+that isBefore you continue make sure that you've closed Minicom! Only 1 program at the
+time can use serial line. from ROM.
+
+Now bring the BMS pin back to `0`. This is important! I took me 2 days to
+discover that the BMS pin must set back to `0` in order to execute most of the
+applets.
+
+Validate that SAM-BA detects the serial port by running:
+
+``` bash
+$ ./sam-ba --port help
+Known ports: serial, j-link
+```
+
+Before you can write anything to the NAND flash the RAM most be initialized.
+This can be done by using the `extram` applet:
+
+``` sam-ba
+$ ./sam-ba -p serial -b sam9xx5-ek -a extram
+Opening serial port 'ttyUSB0'
+Connection opened.
+Compatible device detected: SAM9G25.
+Connection closed.
+```
+
+## Debrick NAND
+Now it's time to to use the `nandflash` applet. Before writing a segment of the
+NAND you should erase the existing content. This can be done using the `erase`
+command. Then you can write a file to the NAND. You can use the either the
+`writeboot` or the `write` command. The first command is for writing
```

Comment:
Weet je ook wat het verschil is tussen die commando's, ik snap dat het voor de bootloader is maar kan ik ook een bootloader schrijven met het "write" command?

Reply:
Nope, helaas niet.